server/aap/formatters/aaprss.py

```diff
@@ -0,0 +1,261 @@
+        "media": "http://search.yahoo.com/mrss/",
+    }
+    item_field: str = "item"
+    item_id_field: str = "id"
+    public_url = "https://aapnews.com.au/news/"
+
+    async def get_db_docs(self, items: List[dict[str, Any]]) -> dict[str, Any]:
+        """
+        Given the docs found by the search, we get the complete version from mongo.
+        :param items:
+        :return: List of documents from mongo
+        """
+        wire_service = WireSearchServiceAsync().service
+        cursor = await wire_service.search(
+            {"_id": {"$in": [d.get("_id") for d in items]}}, use_mongo=True
+        )
+        return {i.id: i.to_dict() async for i in cursor}
+
+    async def get_original_docs(self, db_docs: dict[str, Any]) -> dict[str, Any]:
+        """
+        find the id's of the original published version as the canonical URL is set by the headline of the 1st version
+        :param db_docs:
+        :return:
+        """
+        original_ids = []
+        for db_doc in db_docs.values():
+            if db_doc.get("ancestors"):
+                original_id = (
+                    db_doc.get("ancestors")[0]
+                    if len(db_doc.get("ancestors")) >= 1
+                    else None
+                )
+                if original_id:
+                    original_ids.append(original_id)
+
+        wire_service = WireSearchServiceAsync().service
+        cursor = await wire_service.search(
+            {"_id": {"$in": original_ids}}, use_mongo=True
+        )
+        return {i.id: i.to_dict() async for i in cursor}
+
+    def get_original_item(
+        self, complete_item: dict, original_docs: dict[str, Any]
+    ) -> dict | None:
+        """
+        Given an item, try to find the original version in the original_docs, there is a special case for
+        items that have been corrected, killed or taken down but have only a single version, in that case we need to
+        look it up in the items_versions collection
+        :param complete_item:
+        :param original_docs:
+        :return:
+        """
+        original_item = None
+        ancestors = complete_item.get("ancestors")
+        if ancestors:
+            original_item = original_docs.get(ancestors[0])
+            if not original_item:
+                logger.error(
+                    "original_item not found for {}".format(complete_item.get("_id"))
+                )
+        else:  # handle the kills, takedowns and corrections, they should bot be that common
+            if not complete_item.get("pubstatus") == "usable" or not complete_item.get(
+                "firstpublished"
+            ) == complete_item.get("versioncreated"):
+                req = ParsedRequest()
+                req.sort = (
+                    "[('_current_version', 1)]"  # make sure the oldest version is first
+                )
+                versions = list(
+                    get_resource_service("items_versions").get_from_mongo(
```

Comment:
There is an [async interface](https://github.com/superdesk/superdesk-core/blob/release/3.0-alpha/superdesk/core/resources/service.py#L97) into the versions collection. It uses the [_versions suffix](https://github.com/superdesk/superdesk-core/blob/release/3.0-alpha/superdesk/core/mongo/mongo_manager.py#L254).

I know it's a direct connection to MongoDB, would this work in this scenario?

I know this scenario of killed etc won't happen often, but using old Eve resources does limit our ability to remove Eve in the future (or at least adds to the complexity of removing it). It would be good to use async now if we can, otherwise we know there's a limitation of our current async we need to look at later on